core/store/tx_manager.go

```diff
@@ -691,23 +696,36 @@ func (txm *EthTxManager) handleSafe(
 	return nil
 }
 
-// bumpGas creates a new transaction attempt with an increased gas cost
 func (txm *EthTxManager) bumpGas(tx *models.Tx, attemptIndex int, blockHeight uint64) error {
 	txAttempt := tx.Attempts[attemptIndex]
 
 	originalGasPrice := txAttempt.GasPrice.ToInt()
 	bumpedGasPrice := new(big.Int).Add(originalGasPrice, txm.config.EthGasBumpWei())
 
-	bumpedTxAttempt, err := txm.createAttempt(tx, bumpedGasPrice, blockHeight)
-	if err != nil {
-		return errors.Wrapf(err, "bumpGas from Tx #%s", txAttempt.Hash.Hex())
-	}
+	for {
+		bumpedTxAttempt, err := txm.createAttempt(tx, bumpedGasPrice, blockHeight)
+		if isUnderPricedReplacementError(err) {
+			// Geth rejects bumps smaller than some configurable percentage
+			// with "replacement transaction underpriced" The correct thing to
+			// do in this case is to bump again immediately since this
+			// transaction will never be confirmed otherwise
+			logger.Debugw("gas bump was rejected by ethereum node as underpriced, bumping again",
+				"originalGasPrice", originalGasPrice, "bumpedGasPrice", bumpedGasPrice,
+			)
+			bumpedGasPrice = new(big.Int).Add(bumpedGasPrice, txm.config.EthGasBumpWei())
+			continue
```

Comment:
This looks like a CPU stealing loop if we somehow are unable to get to an appropriate gas price, we should probably put a limit on it.